rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -95,17 +96,17 @@ trait RuleOrNodeReportingServiceImpl extends ReportingService {
       nodeIds <- confExpectedRepo.findCurrentNodeIds(ruleId)
       reports <- findRuleNodeStatusReports(nodeIds, Set(ruleId))
     } yield {
-      RuleStatusReport(ruleId, reports.toIterable)
+      val toKeep = reports.values.flatMap( _._2.filter( _.ruleId == ruleId))
```

Comment:
why are you refiltering ? findRuleNodeStatusReports is already filtering by ruleId